### PR TITLE
guest_os_booting: add new case for backed nvram source

### PR DESCRIPTION
**Pull request by meinaLi**
This case is to verify the ovmf different backed nvram source. Now it only cover the file type.

### CODE REVIEW

**Comment by meinaLi** [comment]
```
# avocado run --vt-type libvirt --test-runner=runner --vt-machine-type q35 guest_os_booting.ovmf_backed_nvram
JOB ID     : e818c6b1d41ffe98e5c93ef0ce57e123f06ea4d3
JOB LOG    : /var/lib/avocado/job-results/job-2023-09-07T23.14-e818c6b/job.log
 (1/3) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_backed_nvram.file_source.default: PASS (35.70 s)
 (2/3) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_backed_nvram.file_source.with_selinux_seclabel: PASS (35.19 s)
 (3/3) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_backed_nvram.file_source.with_dac_seclabel: PASS (47.22 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-09-07T23.14-e818c6b/results.html
JOB TIME   : 118.84 s
```

**Comment by meinaLi** [comment]
Depend on https://github.com/avocado-framework/avocado-vt/pull/3764.

File: libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_backed_nvram.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+from virttest import libvirt_version
+from virttest import virsh
+from virttest.data_dir import get_data_dir
+from virttest.libvirt_xml import vm_xml
+from provider.guest_os_booting import guest_os_booting_base as guest_os
+
+
+def run(test, params, env):
+    """
+    This case is to verify the ovmf  backed nvram.
```
Comment (cliping) [inline libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_backed_nvram.py:23]: Has extra spaces between 'ovmf' and 'backed'.

**Review by cliping — APPROVED** [review]
Others LGTM

File: libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_backed_nvram.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    1) Prepare a guest with related backed nvram elements.
+    2) Start and boot the guest.
+    3) Check the dumpxml and the label if necessary.
+    """
+    def compare_guest_xml():
```
Comment (Yingshun) [inline libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_backed_nvram.py:28]: I think it's better to pass 'vmxml' and 'os_attrs' as arguments of this function to make them as local variables, I think it's easy for others to understand it.

**Review by Yingshun — COMMENTED** [review]
Others LGTM

File: libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_backed_nvram.py (hunk 1 above)
Comment (meinaLi) [reply libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_backed_nvram.py]: Updated.

File: libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_backed_nvram.py (hunk 2 above)
Comment (meinaLi) [reply libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_backed_nvram.py]: Added the arguments to the function compare_guest_xml(vmxml, os_attrs).

File: libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_backed_nvram.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        """
+        os_xml = vmxml.os
+        current_os_attrs = os_xml.fetch_attrs()
+        for key in os_attrs:
+            if key in current_os_attrs:
+                if os_attrs[key] != current_os_attrs[key]:
+                    test.fail("Configured os xml value {} doesn't match the"
+                              " entry {} in guest xml".format(os_attrs[key], current_os_attrs[key]))
+            else:
+                test.fail("Configured os attributes {} aren't existed in guest.".format(key))
```
Comment (Yingshun) [inline libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_backed_nvram.py:40]: :%s/aren't/don't

File: libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_backed_nvram.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        test.log.info("Check the os xml in dumpxml")
+        compare_guest_xml(vmxml, os_attrs)
+        test.log.info("Check the nvram file label in host")
+        label_result = process.run("ls -lZ {}".format(nvram_file)).stdout_text
+        if seclabel_model == "dac":
+            seclabel_label = seclabel_label.replace(":", " ")
+        if seclabel_label in label_result:
+            test.log.info("Get expected nvram file label: {}".format(label_result))
+        else:
+            test.fail("The nvram file label %{} is not expected".format(label_result))
```
Comment (Yingshun) [inline libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_backed_nvram.py:69]: Remove %?

File: libvirt/tests/cfg/guest_os_booting/ovmf_firmware/ovmf_backed_nvram.cfg [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        - file_source:
+            nvram_attrs = {'nvram_attrs': {'template': '${template_path}', 'type': 'file'}}
+            variants:
+                - default:
+                    seclabel_label = "system_u:object_r:svirt_image_t:s0"
+                    nvram_dict = {'nvram_source': {'attrs': {'file': '%s'}}}
+                - with_selinux_seclabel:
+                    seclabel_model = "selinux"
+                    seclabel_label = "system_u:object_r:nvram_image_t:s0"
+                    nvram_dict = {'nvram_source': {'seclabels': [{'label': '${seclabel_label}', 'model': '${seclabel_model}', 'relabel': 'yes'}], 'attrs': {'file': '%s'}}}
```
Comment (Yingshun) [inline libvirt/tests/cfg/guest_os_booting/ovmf_firmware/ovmf_backed_nvram.cfg:21]: These ones(nvram_attrs and nvram_dict) are a bit similar, how about replacing 'nvram_dict' with 'nvram_source'?

File: libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_backed_nvram.py (hunk 3 above)
Comment (meinaLi) [reply libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_backed_nvram.py]: Updated.

File: libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_backed_nvram.py (hunk 4 above)
Comment (meinaLi) [reply libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_backed_nvram.py]: Updated.

File: libvirt/tests/cfg/guest_os_booting/ovmf_firmware/ovmf_backed_nvram.cfg (hunk 5 above)
Comment (meinaLi) [reply libvirt/tests/cfg/guest_os_booting/ovmf_firmware/ovmf_backed_nvram.cfg]: Yes, a little similar. So changed to nvram_source.

**Comment by meinaLi** [comment]
Split the manual case to two cases: test the ovmf file backed nvram and test the ovmf nvram seclable. So update the script here.
```
# avocado run --vt-type libvirt --test-runner=runner --vt-machine-type q35 guest_os_booting.ovmf_backed_nvram
JOB ID     : a507e5b2eeb0bdafd96ef488dbf7757b1be1988e
JOB LOG    : /var/lib/avocado/job-results/job-2023-11-06T01.32-a507e5b/job.log
 (1/1) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_backed_nvram.file: PASS (33.94 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-11-06T01.32-a507e5b/results.html
JOB TIME   : 34.73 s
```

**Review by Yingshun — APPROVED** [review]

**Pull request merged**